### PR TITLE
Use CircleCI 2.0

### PR DESCRIPTION
**Pull request by jaredallard**
Best to stay with the times, pretty sure they're deprecating the 1.0 format now anyways.

When/If this is merged, please _squash_. CI tests suck because you have to commit a lot.

### CODE REVIEW

**Comment by calebboyd** [comment]
Oh hey, the nexe org is setup for Mac builds in circle. I don't know if 2.0 supports them yet. I'm afk right now but can take a look in a bit

**Comment by jaredallard** [comment]
@calebboyd Ahhhh, gotcha, I'll switch it to that

**Comment by jaredallard** [comment]
Fixed :tada: 

Turns out you should probably run `yarn` in CI, haha. LGTM :shipit:

File: .circleci/bootstrap.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+nvs add lts
+nvs link lts
+nvs use default
+echo "using node $(node --version)"
+
+# Install latest yarn version
+npm install -g yarn
+
+echo "using yarn $(yarn --version)"
+npm install -g ts-node
```
Comment (calebboyd) [inline .circleci/bootstrap.sh]: L18 & 19 aren't needed in lieu of L22 ;)

File: package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
   "scripts": {
     "asset-compile": "ts-node tasks/asset-compile",
     "test": "mocha",
-    "fmt": "prettier --parser typescript --no-semi --print-width 100 --single-quote --write \"{src,plugins,tasks}/**/*.ts\"",
+    "fmt": "prettier --parser typescript --no-semi   --print-width 100 --single-quote --write \"{src,plugins,tasks}/**/*.ts\"",
```
Comment (calebboyd) [inline package.json]: I don't think this diff does anything

Comment (jaredallard) [reply package.json]: it did for the caching (yarn didn't run before so we had 32 byte cache entry, had to force it to get a new sha hash

File: .circleci/bootstrap.sh (hunk 1 above)
Comment (jaredallard) [reply .circleci/bootstrap.sh]: Fixed

**Pull request merged**